source/nanoFramework.Hardware.Esp32/TouchPad/TouchPad.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+
+namespace nanoFramework.Hardware.Esp32.TouchPad
+{
+/// <summary>
+/// TouchPad sensor using interrupts
+/// </summary>
+  public sealed class TouchPad : TouchPadBase
+  {
+    public delegate void TouchPadValueChangedEventHandler(
+        Object sender);
+		//TouchPadValueChangedEventArgs e);
+
+
+		/// <summary>
+		/// Simplified constructor which use default settings
+		/// </summary>
+		/// <param name="pinNumber">Valid touch pads pin number</param>
+		public TouchPad(int pinNumber) : this(pinNumber, new TouchPadConfig())
+		{
+		}
+
+		/// <summary>
+		/// Constructor that allow fine tuning of pit settings
+		/// </summary>
+		/// <param name="pinNumber">Valid touch pads pin number</param>
+		/// <param name="config">Configuration settings</param>
+		public TouchPad(int pinNumber, TouchPadConfig config) : base(pinNumber, config)
+    {
+			SetTouchPadTriggerThreshold(config.InterruptThresholdValue);
+		}
+
+		/// <summary>
+		/// The default FSM mode is ‘TOUCH_FSM_MODE_SW’. If you want to use interrupt trigger mode, 
+		/// then set it to ‘TOUCH_FSM_MODE_TIMER’ after calling init function.
+		/// </summary>
+		protected override void SetFsmMode()
+		{
+			if (!base.TouchPadSetFsmMode(TouchFsmMode.TOUCH_FSM_MODE_TIMER))
+				throw new Exception("Failed to set FSM mode.");
```

Comment:
These strings go into the PE, which is loaded in the device, which "wastes" flash space. It's very easy for a deployment to grow a few kBytes with "just" strings.
The best approach (which works perfectly here) is to use the empty exception constructor and _then_ add to the method documentation a list with all possible exceptions that can be thrown from the method. If it makes sense, the reason why and, if it's not much trouble/makes sense the way to fix that on the caller or how to avoid or process.
This would be using the notation (at the bottom of the method comment): `/// <exception cref="EXCEPTION-TYPE-HERE">EXPLANATION-HERE.</exception>`

Reply:
I understood that custom exception classes shall be created and throw like this:
`throw new SettingFsmModeException();`
Please confirm.

Reply:
No need to add a new Exception type. 
Just remove the message from the Exception constructor.
For completeness you can add to the method comment the explanation for why it would throw (like I was trying to explain)